sci-libs/pytorch/pytorch-1.10.0.ebuild

```diff
@@ -0,0 +1,328 @@
+https://github.com/pytorch/QNNPACK/archive/7d2a4e9931a82adc3814275b6219a03e24e36b4c.tar.gz -> QNNPACK-7d2a4e9931a82adc3814275b6219a03e24e36b4c.tar.gz
+https://github.com/shibatch/sleef/archive/e0a003ee838b75d11763aa9c3ef17bf71a725bff.tar.gz -> sleef-e0a003ee838b75d11763aa9c3ef17bf71a725bff.tar.gz
+https://github.com/pytorch/tensorpipe/archive/d2aa3485e8229c98891dfd604b514a39d45a5c99.tar.gz -> tensorpipe-d2aa3485e8229c98891dfd604b514a39d45a5c99.tar.gz
+https://github.com/google/googletest/archive/2fe3bd99.tar.gz -> googletest-2fe3bd99.tar.gz
+https://github.com/google/libnop/archive/aa95422e.tar.gz -> libnop-aa95422e.tar.gz
+https://github.com/libuv/libuv/archive/48e04275332f5753427d21a52f17ec6206451f2c.tar.gz -> libuv-48e04275332f5753427d21a52f17ec6206451f2c.tar.gz
+https://github.com/google/XNNPACK/archive/79cd5f9e18ad0925ac9a050b00ea5a36230072db.tar.gz -> XNNPACK-79cd5f9e18ad0925ac9a050b00ea5a36230072db.tar.gz
+https://github.com/pytorch/kineto/archive/879a203d9bf554e95541679ddad6e0326f272dc1.tar.gz -> kineto-879a203d9bf554e95541679ddad6e0326f272dc1.tar.gz
+https://github.com/driazati/breakpad/archive/7d188f679d4ae0a5bd06408a3047d69ef8eef848.tar.gz -> breakpad-7d188f679d4ae0a5bd06408a3047d69ef8eef848.tar.gz
+https://github.com/mikey/linux-syscall-support/archive/e1e7b0ad8ee99a875b272c8e33e308472e897660.tar.gz -> lss-e1e7b0ad8ee99a875b272c8e33e308472e897660.tar.gz
+"
+
+# git clone git@github.com:pytorch/pytorch.git && cd pytorch
+# git submodules update --init --recursive
+# ${FILESDIR}/get_third_paries
+# cat SRC_URI src_prepare
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~x86 ~amd64"
+
+IUSE="asan blas cuda +fbgemm ffmpeg gflags glog +gloo leveldb lmdb mkldnn mpi namedtensor +nnpack numa +observers opencl opencv +openmp +python +qnnpack redis rocm static test tools zeromq"
+RESTRICT="!test? ( test )"
+REQUIRED_USE="
+	?? ( cuda rocm )
+"
+
+RDEPEND="
+	dev-python/pyyaml[${PYTHON_USEDEP}]
+	blas? ( virtual/blas )
+	cuda? ( dev-libs/cudnn
+		dev-cpp/eigen[cuda] )
+	rocm? ( >=dev-util/hip-4.0.0-r1
+			>=dev-libs/rccl-4
+			>=sci-libs/rocThrust-4
+			>=sci-libs/hipCUB-4
+			>=sci-libs/rocPRIM-4
+			>=sci-libs/miopen-4
+			>=sci-libs/rocBLAS-4
+			>=sci-libs/rocRAND-4
+			>=sci-libs/hipSPARSE-4
+			>=sci-libs/rocFFT-4
+			>=dev-util/roctracer-4 )
+	ffmpeg? ( media-video/ffmpeg )
+	gflags? ( dev-cpp/gflags )
+	glog? ( dev-cpp/glog[gflags] )
+	leveldb? ( dev-libs/leveldb )
+	lmdb? ( dev-db/lmdb )
+	mpi? ( virtual/mpi )
+	opencl? ( dev-libs/clhpp virtual/opencl )
+	opencv? ( media-libs/opencv )
+	python? ( ${PYTHON_DEPS}
+		dev-python/pybind11[${PYTHON_USEDEP}]
+		dev-python/numpy[${PYTHON_USEDEP}]
+		dev-python/protobuf-python:0/30
+	)
+	redis? ( dev-db/redis )
+	zeromq? ( net-libs/zeromq )
+	dev-cpp/eigen
+	dev-libs/protobuf:0/30
+	dev-libs/libuv
+"
+
+#ATen code generation
+BDEPEND="dev-python/pyyaml"
+
+DEPEND="${RDEPEND}
+	test? ( dev-python/pytest[${PYTHON_USEDEP}] )
```

Comment:
We can shorten this by calling `distutils_enable_tests pytest` in global scope. It sets IUSE="test", RESTRICT, and DEPEND automatically.

```suggestion
```